packages/dev/core/src/ShadersWGSL/iblShadowsCdfy.fragment.fx

```diff
@@ -0,0 +1,50 @@
+#define PI 3.1415927
+varying vUV: vec2f;
+
+#ifdef IBL_USE_CUBE_MAP
+var iblSourceSampler: sampler;
+var iblSource: texture_cube<f32>;
+#else
+var iblSourceSampler: sampler;
+var iblSource: texture_2d<f32>;
+#endif
+uniform iblHeight: i32;
+
+#ifdef IBL_USE_CUBE_MAP
+fn equirectangularToCubemapDirection(uv: vec2f) -> vec3f {
+  var longitude: f32 = uv.x * 2.0 * PI - PI;
+  var latitude: f32 = PI * 0.5 - uv.y * PI;
+  var direction: vec3f;
+  direction.x = cos(latitude) * sin(longitude);
+  direction.y = sin(latitude);
+  direction.z = cos(latitude) * cos(longitude);
+  return direction;
+}
+
+fn fetchCube(uv: vec2f) -> f32 {
+  var direction: vec3f = equirectangularToCubemapDirection(uv);
+  return sin(PI * uv.y) * dot(textureSampleLevel(iblSource, iblSourceSampler, direction, 0.0).rgb,
+                               vec3f(0.3, 0.6, 0.1));
+}
+#else
+
+fn fetchPanoramic(Coords: vec2i, envmapHeight: f32) -> f32 {
+  return sin(PI * ( f32(Coords.y) + 0.5) / envmapHeight) *
+         dot(textureLoad(iblSource, Coords, 0).rgb,  vec3f(0.3, 0.6, 0.1));
+}
+#endif
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+  var coords: vec2i =  vec2i(i32(input.position.x), i32(input.position.y));
```

Comment:
shader preprocessor can't handle this expression see [here](https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Engines/WebGPU/webgpuShaderProcessorsWGSL.ts#L280)
```suggestion
  var coords: vec2i =  vec2i(fragmentInputs.position.xy);
```